bootstrap/internal/controllers/rke2config_controller.go

```diff
@@ -278,6 +278,14 @@ type Scope struct {
 	ControlPlane         *controlplanev1.RKE2ControlPlane
 }
 
+func (s *Scope) getDesiredVersion() string {
+	if s.Machine.Spec.Version != nil && bsutil.IsRKE2Version(*s.Machine.Spec.Version) {
+		return *s.Machine.Spec.Version
+	}
+
+	return s.Config.Spec.AgentConfig.Version
```

Comment:
I updated #196 to remind us to remove the `AgentConfig.Version` field in the next version bump